DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -2629,6 +2711,46 @@ internal async Task OnStageInstanceDeleteAsync(DiscordStageInstance instance)
 
     #region Misc
 
+    internal async Task OnApplicationAuthorizedAsync(DiscordWebhookEventBody body)
+    {
+        ApplicationAuthorizedPayload payload = body.Data.ToDiscordObject<ApplicationAuthorizedPayload>();
+
+        DiscordGuild? guild = payload.Guild;
+        DiscordUser user = payload.User;
+
+        UpdateUserCache(user);
+
+        if (guild is not null)
+        {
+            if (this.guilds.TryGetValue(guild.Id, out DiscordGuild? cachedGuild))
+            {
+                guild = cachedGuild;
+            }

```

Comment:
@VelvetToroyashi did as you suggested at https://canary.discord.com/channels/379378609942560770/379386901725052928/1299821870883012710 , however, it does also occur to me that if the guild is something we're being authorized into, it's very likely we *do not* already have it cached - in fact, i would be surprised if we did have it cached - so the usefulness of this code may... *vary*

Reply:
The field should be nullable for this reason, no? Preventing people from footgunning themselves > Discord's documentation

Reply:
it's already nullable because it's not guaranteed to be present on user auth, anyway